delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Resolvers.scala

```diff
@@ -0,0 +1,267 @@
+            .transform
+        }
+      // Create a cross-project resolver
+      case (Initial, c: CreateCrossProjectResolver)                =>
+        for {
+          _   <- validateIdentities(c.identities)
+          now <- instant
+        } yield c
+          .into[CrossProjectResolverCreated]
+          .withFieldConst(_.rev, 1L)
+          .withFieldConst(_.instant, now)
+          .withFieldConst(_.subject, caller.subject)
+          .transform
+
+      /**
+        * Updating a resolver
+        */
+      // Update a non existing resolver
+      case (Initial, c: UpdateResolver)                            =>
+        IO.raiseError(ResolverNotFound(c.id, c.project))
+      // Invalid revision has been provided
+      case (s: Current, c: UpdateResolver) if c.rev != s.rev       =>
+        IO.raiseError(IncorrectRev(c.rev, s.rev))
+
+      // Update a in-project resolver
+      case (s: InProjectCurrent, c: UpdateInProjectResolver)       =>
+        instant.map { now =>
+          c.into[InProjectResolverUpdated]
+            .withFieldConst(_.rev, s.rev + 1)
+            .withFieldConst(_.instant, now)
+            .withFieldConst(_.subject, caller.subject)
+            .transform
+        }
+      // Update a cross-project resolver
+      case (s: CrossProjectCurrent, c: UpdateCrossProjectResolver) =>
+        for {
+          _   <- validateIdentities(c.identities)
+          now <- instant
+        } yield c
+          .into[CrossProjectResolverUpdated]
+          .withFieldConst(_.rev, s.rev + 1)
+          .withFieldConst(_.instant, now)
+          .withFieldConst(_.subject, caller.subject)
+          .transform
+
+      // Attempt to transform an in-project to a cross-project
+      case (_: InProjectCurrent, c: UpdateCrossProjectResolver)    =>
+        IO.raiseError(DifferentResolverType(c.id, crossProjectType, inProjectType))
+      // Attempt to transform a cross-project to an in-project
+      case (_: CrossProjectCurrent, c: UpdateInProjectResolver)    =>
+        IO.raiseError(DifferentResolverType(c.id, inProjectType, crossProjectType))
+
+      /**
+        * Deprecating a resolver
+        */
+      // Resolver can't be found
+      case (Initial, c: DeprecateResolver)                         =>
+        IO.raiseError(ResolverNotFound(c.id, c.project))
+      // Invalid revision
+      case (s: Current, c: DeprecateResolver) if c.rev != s.rev    =>
+        IO.raiseError(IncorrectRev(c.rev, s.rev))
+      case (s: Current, _: DeprecateResolver) if s.deprecated      =>
+        IO.raiseError(ResolverIsDeprecated(s.id))
+      case (s: Current, c: DeprecateResolver)                      =>
+        instant.map { now =>
+          c.into[ResolverDeprecated]
+            .withFieldConst(_.rev, s.rev + 1)
+            .withFieldConst(_.instant, now)
+            .withFieldConst(_.subject, caller.subject)
+            .transform
+        }
+
+    }
+
+    isOpen(command.project) >> stateMachine
```

Comment:
This check should be done in the `Resolvers` and not in the state machine, since otherwise in some cases you are gonna ask 2 times for the project (The method Resolvers.create(project, resolverFields) needs to make the same check to fetch the project in order to then compute the id)